### PR TITLE
feat: [CDB-227]: enforce dashboards view and edit roles

### PR DESCRIPTION
**Pull request by sean-harness**
##### Summary:

CDB-227 will enforce the Dashboard View and Edit roles within the app.
The main changes happening to the UI as part of this change are
* Support displaying meaningful error messages
* Enforce roles per folder
* Include OOTB folder when assigning a resource group

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by sean-harness** [comment]
retrigger titlecheck

**Comment by sean-harness** [comment]
retrigger titlecheck

**Comment by sean-harness** [comment]
retrigger sonar

File: src/modules/45-dashboards/pages/home/HomePage.tsx [hunk 1]
```diff
@@ -215,14 +212,22 @@ const HomePage: React.FC = () => {
...
+      showError(e?.data?.responseMessages || getString('dashboards.deleteDashboard.failed'))
+    }
   }
 
   return (
-    <Page.Body loading={loading || deleting} error={(error?.data as ErrorResponse)?.responseMessages}>
+    <Page.Body
+      loading={loading || deleting}
+      error={(error?.data as ErrorResponse)?.responseMessages}
+      retryOnError={() => refetchDashboards()}
```
Comment (Harness-JKhattak) [inline src/modules/45-dashboards/pages/home/HomePage.tsx:229]: Would this work the same if you just did:

`retryOnError={refetchDashboards}`

**Review by Harness-JKhattak — APPROVED** [review]

File: src/modules/45-dashboards/pages/home/DashboardForm.tsx [hunk 2]
```diff
@@ -58,26 +59,22 @@ const DashboardForm: React.FC<DashboardFormProps> = ({
...
+  const { accountId, folderId } = useParams<DashboardPathProps>()
 
   const {
     data: foldersList,
     error,
     loading: isFoldersLoading
-  } = useGetFolder({
-    queryParams: { accountId }
+  } = useGetFolders({
+    queryParams: { accountId, page: 1, pageSize: 100 }
```
Comment (abhinavrastogi-harness) [inline src/modules/45-dashboards/pages/home/DashboardForm.tsx:70]: no pagination support?

File: src/modules/45-dashboards/pages/home/CloneDashboardForm.tsx [hunk 3]
```diff
@@ -42,7 +42,7 @@ const CloneDashboardForm: React.FC<CloneDashboardFormProps> = ({ hideModal, relo
       const response = await cloneDashboard({ ...completedFormData, dashboardId })
       onSuccess(response)
     } catch (e) {
-      modalErrorHandler?.showDanger(getString('dashboards.cloneDashboardModal.submitFail'))
+      modalErrorHandler?.showDanger(e?.data?.responseMessages || getString('dashboards.cloneDashboardModal.submitFail'))
```
Comment (abhinavrastogi-harness) [inline src/modules/45-dashboards/pages/home/CloneDashboardForm.tsx:45]: use `getErrorInfoFromErrorObject` util function instead

File: src/modules/45-dashboards/pages/folders/useDeleteFolder.tsx [hunk 4]
```diff
@@ -43,8 +43,8 @@ const useDeleteFolder = (folder: FolderModel | undefined, onSuccess: () => void)
             showSuccess(getString('dashboards.deleteFolder.success'))
             onSuccess()
           }
-        } catch (error: unknown | any) {
-          showError(error?.data?.responseMessages)
+        } catch (error) {
+          showError(error?.data?.responseMessages || getString('dashboards.deleteFolder.failed'))
```
Comment (abhinavrastogi-harness) [inline src/modules/45-dashboards/pages/folders/useDeleteFolder.tsx:47]: use `shouldShowError` and `getErrorInfoFromErrorObject`

**Review by abhinavrastogi-harness — APPROVED** [review]

File: src/modules/45-dashboards/pages/home/DashboardForm.tsx (hunk 2 above)
Comment (sean-harness) [reply src/modules/45-dashboards/pages/home/DashboardForm.tsx:70]: This is shown in an input box dropdown, so normal pagination would not be possible.

**Pull request merged**